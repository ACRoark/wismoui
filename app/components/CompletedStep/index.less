@import (reference) '../../styles/themes/index.less';

@completed-step-cls-prefix: ~'@{dte-prefix}-completed-step';

.@{completed-step-cls-prefix} {
  display: flex;

  .icon {
    align-items: center;
    border-radius: 50%;
    display: flex;
    height: 31px;
    justify-content: center;
    padding: 3px 0 0 0;
    transition: background-color 0.3s, border-color 0.3s;
    width: 31px;
  }

  &-icon-wrapper {
    display: inline-flex;
    padding-right: 15px;

    .completed {
      background: @dte-white;
      border: 2px solid @dte-green-darker;
      color: @dte-green-darker;
    }
  }

  &-message {
    color: @dte-green-darker;
    font-size: @dte-wismo-font-size-2;
    font-weight: @dte-wismo-font-weight-semi-bold;
    vertical-align: sub;
  }
}
