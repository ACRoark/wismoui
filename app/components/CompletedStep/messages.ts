import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.completed-step';

export default defineMessages({
  MIMO_START: {
    id: `${scope}.MIMO_START`,
  },
  MIMO_STOP: {
    id: `${scope}.MIMO_STOP`,
  },
});
