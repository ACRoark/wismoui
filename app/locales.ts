export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
