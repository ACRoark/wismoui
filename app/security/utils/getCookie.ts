const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
