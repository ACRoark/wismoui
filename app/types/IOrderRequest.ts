import IOrderRequestBase from './IOrderRequestBase';
import IOrderRequestStatusUpdate from './IOrderRequestStatusUpdate';
import IServiceOrderEvent from './IServiceOrderEvent';
import IVerificationEvent from './IVerificationEvent';

interface IOrderRequest extends IOrderRequestBase {
  contactPhoneNumber: string;
  orderRequestStatusUpdates: IOrderRequestStatusUpdate[];
  serviceOrderEvents: IServiceOrderEvent[];
  verificationEvents: IVerificationEvent[];
}

export default IOrderRequest;
