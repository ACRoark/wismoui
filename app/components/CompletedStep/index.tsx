import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import Address from 'components/Address';
import DateStamp from 'components/DateStamp';
import CompletedIcon from 'components/Icons/CompletedIcon';
import IAddress from 'types/IAddress';

import messages from './messages';

import './index.less';

interface CompletedStepProps {
  address: IAddress;
  completeDate: string;
  requestType: string;
}

const CompletedStep = (props: CompletedStepProps): React.ReactElement => {
  const { address, completeDate, requestType } = props;

  return (
    <div className="dte-wismo-completed-step">
      <div className="dte-wismo-completed-step-icon-wrapper">
        <CompletedIcon />
      </div>
      <h4>
        <span className="dte-wismo-completed-step-message">
          <FormattedMessage
            {...messages[requestType]}
            values={{
              address: <Address address={address} format="street" />,
              date: <DateStamp value={completeDate} />,
            }}
          />
        </span>
      </h4>
    </div>
  );
};

export default CompletedStep;
