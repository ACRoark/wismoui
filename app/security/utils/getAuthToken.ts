import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
