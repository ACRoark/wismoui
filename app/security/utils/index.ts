import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
