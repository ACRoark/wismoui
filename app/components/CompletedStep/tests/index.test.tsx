import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import fakeServiceRequest from 'components/ServiceSummary/tests/fakeServiceRequest';
import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import IOrderRequest from 'types/IOrderRequest';

import CompletedStep from '../index';

const serviceRequestWithAwaitBilling: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestStatusUpdates: [
    {
      createdAtDate: '20200319',
      createdAtTime: '131530',
      orderRequestStatus: 'AWAITING_BILLING',
    },
  ],
  products: [
    {
      docId: '100003441473',
      isCanceled: true,
      processedAtDate: '20201221',
      productType: 'EFR_D1_1',
      serviceDate: '20201227',
    },
    {
      docId: '100003441473',
      isCanceled: false,
      processedAtDate: '20201221',
      productType: 'EFR_D1_1',
      serviceDate: '20201222',
    },
    {
      docId: '100003441473',
      isCanceled: false,
      processedAtDate: '20201221',
      productType: 'EFR_D1_1',
      serviceDate: '20201229',
    },
  ],
  serviceOrderEvents: [
    {
      appointment: {
        date: '2020-04-19',
        slotType: 'AN',
      },
      latestStatus: {
        createdAtDate: '20200418',
        createdAtTime: '131530',
        serviceOrderEventStatus: 'COMPLETED',
      },
      product: 'GCC_GS_1H',
      serviceOrderCategory: null,
      serviceOrderId: 'abc123',
    },
  ],
};

const serviceRequestWithAwaitingBillingAndLine2InAddress: IOrderRequest = {
  ...serviceRequestWithAwaitBilling,
  address: {
    city: 'Anytown',
    line1: '222 Second Rd',
    line2: 'Unit #6',
    state: 'MI',
    zip: '45678-1234',
  },
};
const awaitBillingCompleteDate = '20200418';
const line2CompleteDate = '20200418';

describe('CompletedStep', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered in ${language}`, (): void => {
      const spy = jest.spyOn(global.console, 'error');

      renderWithIntl(
        <CompletedStep
          address={serviceRequestWithAwaitBilling.address}
          completeDate={awaitBillingCompleteDate}
          requestType={serviceRequestWithAwaitBilling.orderRequestType}
        />,
        language,
      );
      expect(spy).not.toHaveBeenCalled();
    });

    // tslint:disable-next-line: max-line-length
    it(`should display a green checkbox and message text including a 1-line address when service request status is AWAITING_BILLING and language is ${language}`, (): void => {
      const tree = createSnapshotWithIntl(
        <CompletedStep
          address={serviceRequestWithAwaitBilling.address}
          completeDate={awaitBillingCompleteDate}
          requestType={serviceRequestWithAwaitBilling.orderRequestType}
        />,
        language,
      );

      expect(tree).toMatchSnapshot();
    });

    it(`should display a green checkbox and message text including a 2-line address when service request status is AWAITING_BILLING, address.line2 is NOT NULL and language is ${language}`, (): void => {
      const tree = createSnapshotWithIntl(
        <CompletedStep
          address={serviceRequestWithAwaitingBillingAndLine2InAddress.address}
          completeDate={line2CompleteDate}
          requestType={serviceRequestWithAwaitingBillingAndLine2InAddress.orderRequestType}
        />,
        language,
      );

      expect(tree).toMatchSnapshot();
    });

    it(`should display the service date from the products array when it renders and language is ${language}`, (): void => {
      const tree = createSnapshotWithIntl(
        <CompletedStep
          address={serviceRequestWithAwaitBilling.address}
          completeDate={serviceRequestWithAwaitBilling.products[2].serviceDate}
          requestType={serviceRequestWithAwaitBilling.orderRequestType}
        />,
        language,
      );

      expect(tree).toMatchSnapshot();
    });
  });
});
