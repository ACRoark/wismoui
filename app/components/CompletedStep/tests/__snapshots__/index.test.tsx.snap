// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedStep should display a green checkbox and message text including a 1-line address when service request status is AWAITING_BILLING and language is en 1`] = `
<div
  className="dte-wismo-completed-step"
>
  <div
    className="dte-wismo-completed-step-icon-wrapper"
  >
    <svg
      className="completed icon"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
        fillRule="evenodd"
        stroke="currentColor"
        strokeWidth="1px"
      />
    </svg>
  </div>
  <h4>
    <span
      className="dte-wismo-completed-step-message"
    >
      Your order to start service at 
      <address
        className="dte-wismo-address"
      >
        123 Main
      </address>
       was completed on 
      04/18/20
      .
    </span>
  </h4>
</div>
`;

exports[`CompletedStep should display a green checkbox and message text including a 1-line address when service request status is AWAITING_BILLING and language is es 1`] = `
<div
  className="dte-wismo-completed-step"
>
  <div
    className="dte-wismo-completed-step-icon-wrapper"
  >
    <svg
      className="completed icon"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
        fillRule="evenodd"
        stroke="currentColor"
        strokeWidth="1px"
      />
    </svg>
  </div>
  <h4>
    <span
      className="dte-wismo-completed-step-message"
    >
      Su pedido para iniciar el servicio en 
      <address
        className="dte-wismo-address"
      >
        123 Main
      </address>
       se completó el 
      18/04/20
      .
    </span>
  </h4>
</div>
`;

exports[`CompletedStep should display a green checkbox and message text including a 2-line address when service request status is AWAITING_BILLING, address.line2 is NOT NULL and language is en 1`] = `
<div
  className="dte-wismo-completed-step"
>
  <div
    className="dte-wismo-completed-step-icon-wrapper"
  >
    <svg
      className="completed icon"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
        fillRule="evenodd"
        stroke="currentColor"
        strokeWidth="1px"
      />
    </svg>
  </div>
  <h4>
    <span
      className="dte-wismo-completed-step-message"
    >
      Your order to start service at 
      <address
        className="dte-wismo-address"
      >
        222 Second Rd, Unit #6
      </address>
       was completed on 
      04/18/20
      .
    </span>
  </h4>
</div>
`;

exports[`CompletedStep should display a green checkbox and message text including a 2-line address when service request status is AWAITING_BILLING, address.line2 is NOT NULL and language is es 1`] = `
<div
  className="dte-wismo-completed-step"
>
  <div
    className="dte-wismo-completed-step-icon-wrapper"
  >
    <svg
      className="completed icon"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
        fillRule="evenodd"
        stroke="currentColor"
        strokeWidth="1px"
      />
    </svg>
  </div>
  <h4>
    <span
      className="dte-wismo-completed-step-message"
    >
      Su pedido para iniciar el servicio en 
      <address
        className="dte-wismo-address"
      >
        222 Second Rd, Unit #6
      </address>
       se completó el 
      18/04/20
      .
    </span>
  </h4>
</div>
`;

exports[`CompletedStep should display the service date from the products array when it renders and language is en 1`] = `
<div
  className="dte-wismo-completed-step"
>
  <div
    className="dte-wismo-completed-step-icon-wrapper"
  >
    <svg
      className="completed icon"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
        fillRule="evenodd"
        stroke="currentColor"
        strokeWidth="1px"
      />
    </svg>
  </div>
  <h4>
    <span
      className="dte-wismo-completed-step-message"
    >
      Your order to start service at 
      <address
        className="dte-wismo-address"
      >
        123 Main
      </address>
       was completed on 
      12/29/20
      .
    </span>
  </h4>
</div>
`;

exports[`CompletedStep should display the service date from the products array when it renders and language is es 1`] = `
<div
  className="dte-wismo-completed-step"
>
  <div
    className="dte-wismo-completed-step-icon-wrapper"
  >
    <svg
      className="completed icon"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
        fillRule="evenodd"
        stroke="currentColor"
        strokeWidth="1px"
      />
    </svg>
  </div>
  <h4>
    <span
      className="dte-wismo-completed-step-message"
    >
      Su pedido para iniciar el servicio en 
      <address
        className="dte-wismo-address"
      >
        123 Main
      </address>
       se completó el 
      29/12/20
      .
    </span>
  </h4>
</div>
`;
